### PR TITLE
Only load global step when fitting

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Changes the behavior of loading the global_step. 

**Current:** 
Global step gets reloaded regardless of whether you call `Trainer.fit(ckpt_path=...)` or `Trainer.validate(ckpt_path=...), ...`

**Now:**
Global step will only restore its value when calling `Trainer.fit(ckpt_path=...)`. Changed test reflects this behavior. 

This is a breaking change, but aligns the behavior as expected for consistency with the rest of our reloading logic.

Discovered in #15500 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @carmocca @rohitgr7

### CODE REVIEW

File: src/pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 1]
```diff
@@ -343,17 +343,18 @@ def restore_loops(self) -> None:
 
         # set the `global_step` value for checkpoints before v1.6 without the progress tracking state.
         # it will be overwritten by the loop's state if it was also saved
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:345]: This comment should be inside the `if`

**Review by carmocca — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 2]
```diff
@@ -341,19 +341,20 @@ def restore_loops(self) -> None:
...
-            batch_loop.optimizer_loop.optim_progress.optimizer.step.total.completed = self._loaded_checkpoint[
-                "global_step"
-            ]
-        else:
-            batch_loop.manual_loop.optim_step_progress.total.completed = self._loaded_checkpoint["global_step"]
-
-        # set the `current_epoch` value for checkpoints before v1.6 without the progress tracking state.
-        # it will be overwritten by the loop's state if it was also saved
-        fit_loop.epoch_progress.current.completed = self._loaded_checkpoint["epoch"]
+        if self.trainer.state.fn == TrainerFn.FITTING:
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:344]: Reviewer: all we did here is add the if statement.

File: src/pytorch_lightning/CHANGELOG.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - From now on, Lightning Trainer and `LightningModule.load_from_checkpoint` automatically upgrade the loaded checkpoint if it was produced in an old version of Lightning ([#15237](https://github.com/Lightning-AI/lightning/pull/15237))
 
--
+- `Trainer.{validate,test,predict}(ckpt_path=...)` no longer restores the `Trainer.global_step` value from the checkpoints - From now on, only `Trainer.fit` will restore this value ([#15532](https://github.com/Lightning-AI/lightning/pull/15532))
```
Comment (rohitgr7) [inline src/pytorch_lightning/CHANGELOG.md:24]:
```suggestion
- `Trainer.{validate,test,predict}(ckpt_path=...)` no longer restores the `Trainer.global_step` and `trainer.current_epoch` value from the checkpoints - From now on, only `Trainer.fit` will restore this value ([#15532](https://github.com/Lightning-AI/lightning/pull/15532))
```

**Review by rohitgr7 — APPROVED** [review]

**Review by otaj — APPROVED** [review]
